templates/ingress.yaml

```diff
@@ -17,7 +17,7 @@ spec:
   tls:
     - hosts:
         - {{ .Values.ingress.host | quote }}
-      secretName: {{ .Values.ingress.host | replace "." "-" | quote }}
+      secretName: {{ .Values.ingress.tls.secret | default (.Values.ingress.host | replace "." "-") | quote }}
```

Comment:
By default, tls is null, so here is a place for NP.E. Unfortunately, test framework is not good atm, but I believe as soon as you uncomment `tls: null` in values - the template will fail.

Reply:
i put this in values:
```
ingress: 
  # # ingress host
  host: www.cmak.mycompany
  #
  # # ingress path
  path: /
  #
  # # Enable TLS configuration for the hostname defined at ingress.host
  # # secret name will be "${ingress.host.replace(".", "-")}"
  tls: null
  #   # secret: null
  #
  # # optional ingress annotations
  # annotations: {}
  #
  # # optional ingress labels
  # labels: {}
  ```
 and ran: ```helm template -s templates/ingress.yaml . ```
 
 i got:
 ```
 ---
# Source: cmak-operator/templates/ingress.yaml
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: "cmak"
  labels:
    app.kubernetes.io/managed-by: "Helm"
    app.kubernetes.io/version: "3.0.0.5-7e7a22e"
    helm.sh/chart: "cmak-operator-0.0.0"
    app.kubernetes.io/name: "cmak"
    app.kubernetes.io/instance: "RELEASE-NAME"
spec:
  rules:
    - host: "www.cmak.mycompany"
      http:
        paths:
          - path: "/"
            backend:
              serviceName: "cmak"
              servicePort: 9000
 ```

Reply:
Well, that's strange but I cant check it myself atm. Let's keep as it is.